crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -144,7 +161,104 @@ impl AuthorityStorePruner {
+        effects_to_prune: &Vec<TransactionEffects>,
+        metrics: Arc<AuthorityStorePruningMetrics>,
+    ) -> anyhow::Result<()> {
+        let _scope = monitored_scope("EffectsLivePruner");
+
+        let mut perpetual_batch = perpetual_db.objects.batch();
+        let transactions = checkpoint_content_to_prune
+            .iter()
+            .flat_map(|content| content.iter().map(|tx| tx.transaction));
+        for transaction_digest in transactions {
+            if let Some(next_digest) = transaction_digest.next_lexicographical() {
+                perpetual_batch.delete_range(
+                    &perpetual_db.transactions,
+                    &transaction_digest,
+                    &next_digest,
+                )?;
+                perpetual_batch.delete_range(
+                    &perpetual_db.executed_effects,
+                    &transaction_digest,
+                    &next_digest,
+                )?;
+                perpetual_batch.delete_range(
+                    &perpetual_db.executed_transactions_to_checkpoint,
+                    &transaction_digest,
+                    &next_digest,
+                )?;
+            }
+        }
+
+        for effects in effects_to_prune {
+            let effects_digest = effects.digest();
+            if let Some(next_digest) = effects.digest().next_lexicographical() {
+                perpetual_batch.delete_range(
+                    &perpetual_db.effects,
+                    &effects_digest,
+                    &next_digest,
+                )?;
+            }
+            if let Some(event_digest) = effects.events_digest() {
+                if let Some(next_digest) = event_digest.next_lexicographical() {
+                    perpetual_batch.delete_range(
+                        &perpetual_db.events,
+                        &(*event_digest, 0),
+                        &(next_digest, 0),
+                    )?;
+                }
+            }
+        }
+
+        let mut checkpoints_batch = checkpoint_db.certified_checkpoints.batch();
+
+        for checkpoint_content in checkpoint_content_to_prune {
+            let content_digest = *checkpoint_content.digest();
+            if let Some(next_digest) = content_digest.next_lexicographical() {
+                checkpoints_batch.delete_range(
+                    &checkpoint_db.checkpoint_content,
+                    &content_digest,
+                    &next_digest,
+                )?;
+                checkpoints_batch.delete_range(
+                    &checkpoint_db.checkpoint_sequence_by_contents_digest,
+                    &content_digest,
+                    &next_digest,
+                )?;
+            }
+        }
+        for checkpoint_digest in checkpoints_to_prune {
+            if let Some(next_digest) = checkpoint_digest.next_lexicographical() {
+                checkpoints_batch.delete_range(
+                    &checkpoint_db.checkpoint_by_digest,
```

Comment:
I think we may need to keep this - this seems to be the table used by state sync